packages/common/src/content/fetchItemJobRecords.ts

```diff
@@ -0,0 +1,72 @@
+import HubError from "../HubError";
+import {
+  IHubDownloadJobRecord,
+  IHubJobRecord,
+  IHubJobRecordRequestOptions,
+  JobRecordType,
+} from "./types";
+
+/**
+ * Fetches job records for a specific item.
+ *
+ * @param id - The ID of the item.
+ * @param options - The options for fetching job records.
+ * @returns A promise that resolves to an array of job records.
+ * @throws {HubError} If any unimplemented options are provided.
+ */
+export async function fetchItemJobRecords(
+  id: string,
+  options: IHubJobRecordRequestOptions
+): Promise<IHubJobRecord[]> {
+  const unimplementedOptions: Array<keyof IHubJobRecordRequestOptions> = [
+    "types",
+    "statuses",
+  ];
+  if (unimplementedOptions.some((option) => options[option])) {
+    throw new HubError(
+      "fetchItemJobRecords",
+      `The following options are not yet implemented: ${unimplementedOptions.join(
+        ", "
+      )}`
+    );
+  }
+
+  const { context } = options;
+  const requestUrl = `${
+    context.hubUrl
+  }/api/download/v1/items/${id}/errors${getQueryString(options)}`;
+
+  const { errors: rawErrors } = await fetch(requestUrl).then((response) =>
+    response.json()
+  );
+  // TODO: account for additional row types once the API supports them
```

Comment:
Do we have a follow-up ticket for this?

Reply:
Good question! We don't have a specific ticket, but the epic around improving the error logs table will necessitate these changes.